src/test/resources/com/jpinpoint/perf/lang/java/ruleset/common/xml/MDCPutWithoutRemove.xml

```diff
@@ -156,4 +156,114 @@ public class TracingInterceptor extends HandlerInterceptorAdapter {
      ]]></code>
     </test-code>
 
+    <test-code>
+        <description>no violation: Issue 326, doFinally </description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import org.slf4j.MDC;
+import org.springframework.security.web.server.authentication.ServerAuthenticationConverter;
+import reactor.core.publisher.Mono;
+import org.springframework.web.server.ServerWebExchange;
+
+class Issue326 {
+        private static final String MDC_REQUEST_ID_KEY = "Request-id";
+        private final ServerAuthenticationConverter authenticationConverter;
+
+       public Mono<Void> filter(ServerWebExchange serverWebExchange) {
```

Comment:
I have some questions: 
- is doFinally always called, like `finally` in Java?
- is doFinally lambda executed on some thread, or is the MDC available on the doFinally thread? The MDC should be copied and also be cleared in proper way? MDC and reactor are no friends...

Googling for this shows similar questions and (it seems) no simple answers...

Reply:
Yeah, good question. I assumed the requestor is correct in how he uses it.
I see in javadoc: doFinally does not operate by default on a particular [Scheduler](https://reactivex.io/RxJava/2.x/javadoc/2.0.8/io/reactivex/Scheduler.html).

Reply:
I am asking the requestor

Reply:
Zeljko is not sure on second thoughts. He will do some tests to find out.